src/encode_like.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019 Parity Technologies
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::Encode;
+
+/// A marker trait that tells the compiler that two types encode to the same representation.
+///
+/// E.g. `Vec<u8>` has the same encoded representation as `&[u8]`.
+pub trait EncodeLike<T = Self>: Sized + Encode {}
```

Comment:
actually we bound Encode for T here instead of whole type.

maybe better both actually